### PR TITLE
[11.0]]add] account_lock_to_date

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
This module allows to set a Period and Fiscal year Locking end dates. This
will prevent users from posting journal entries on a date after the defined
period or fiscal year end date.

Usage
=====

To set a new lock to dates, go to *Invoicing / Adviser / Actions / Update accounting lock to dates*.

A user without an Adviser group will not be able to post or update posted
journal entries on the date "Lock To Date for Non-Advisers" or after.

A user that has an Adviser group will not be able to post or update posted
journal entries on the date "Lock To Date" or after.

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@jarroyomorales can you review this one please?

File: account_lock_to_date/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2017 ACSONE SA/NV
```
Comment (etobella) [inline account_lock_to_date/__manifest__.py]: It was done by ACSONE?

**Review by etobella — APPROVED** [review]
LGTM :+1: 
Just a comment about the copyright

**Review by jarroyomorales — APPROVED** [review]
LGTM :+1:

**Comment by jarroyomorales** [comment]
Check Travis

**Comment by pedrobaeza** [comment]
What's the difference with https://github.com/OCA/account-financial-tools/tree/11.0/account_lock_date_update?

**Comment by JordiBForgeFlow** [comment]
@pedrobaeza Companies using systems where periods are defined will only allow posting on a defined period, that includes a start and and end date. Odoo currently allows only to lock the start date, but not the end.

This module allows you to set the a lock date IN THE FUTURE. That way you are able to define a complete locking date range.

For example, a company does not want anyone to post journal entries >= 2020 and < 2019

Or, you don't want users to post journal entries with date <> this month. At the end of the month the adviser updates the Period Lock Date to allow users to enter journal entries in the following month.

**Comment by pedrobaeza** [comment]
OK, got it. Thanks for the explanation.

File: account_lock_to_date/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2019 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Account Lock To Date',
+    'summary': """
+        Allos to set an account lock date in the future.""",
```
Comment (nikul-serpentcs) [inline account_lock_to_date/__manifest__.py]: Allos --> allows
@jbeficent  Typo

File: account_lock_to_date/models/res_company.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                    _('You cannot define stricter conditions on advisors '
+                      'than on users. Please make sure that the lock date '
+                      'on advisor is set after the lock date for users.'))
+
+    @api.multi
+    def _validate_fiscalyear_lock(self, values):
+        res = super()._validate_fiscalyear_lock(values)
+        if values.get('fiscalyear_lock_to_date'):
+            nb_draft_entries = self.env['account.move'].search([
+                ('company_id', 'in', [c.id for c in self]),
```
Comment (nikul-serpentcs) [inline account_lock_to_date/models/res_company.py]: here we direct use  `self.ids`

`('company_id', 'in', self.ids),`

File: account_lock_to_date/wizards/account_update_lock_to_date.xml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2017 ACSONE SA/NV
```
Comment (kittiu) [inline account_lock_to_date/wizards/account_update_lock_to_date.xml]: Not sure this should be Eficient.

File: account_lock_to_date/wizards/account_update_lock_to_date.xml [hunk 5]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (kittiu) [inline account_lock_to_date/wizards/account_update_lock_to_date.xml:1]: @jbeficent , just asking for this. Do we actually need this xml tag for view? Or we don't need anymore as for v12.

**Comment by etobella** [comment]
@jbeficent Can you check the last comments? I think this is ready to merge :smile:

File: account_lock_to_date/wizards/account_update_lock_to_date.xml (hunk 4 above)
Comment (JordiBForgeFlow) [reply account_lock_to_date/wizards/account_update_lock_to_date.xml]: @kittiu fixed!

File: account_lock_to_date/wizards/account_update_lock_to_date.xml (hunk 5 above)
Comment (JordiBForgeFlow) [reply account_lock_to_date/wizards/account_update_lock_to_date.xml:1]: @kittiu I can see that it's everywhere, also in v12.

**Pull request merged**